.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,49 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.245.2/containers/docker-existing-dockerfile
+{
+    "name": "IREE Development Container",
+    "dockerComposeFile": [
+        "docker-compose.yml"
+    ],
+    "service": "iree-dev",
+    "remoteUser": "iree",
+    "workspaceFolder": "/workspaces/iree",
+    "customizations": {
+        "vscode": {
+            "extensions": [
+                // LLVM & MLIR
+                "llvm-vs-code-extensions.vscode-clangd",
+                "llvm-vs-code-extensions.vscode-mlir",
+                "jakob-erzar.llvm-tablegen",
+                // CPP
+                "ms-vscode.cpptools-extension-pack",
+                "akiramiyakoda.cppincludeguard",
+                // Python
+                "ms-python.python",
+                // Build Tools
+                "ms-vscode.makefile-tools",
+                "ms-vscode.cmake-tools",
+                "BazelBuild.vscode-bazel",
```

Comment:
Nice, thanks for including these.

We have some common settings for various extensions that we could put here too (https://code.visualstudio.com/docs/devcontainers/containers#_default-container-specific-settings)

Here's a somewhat recent dump of my VSCode config for the project: https://gist.github.com/ScottTodd/fb350ccdc89b6d4b8ad10ee8506ddfa7

Notable settings:
```json
    "cmake.generator": "Ninja",
    "cmake.preferredGenerators": [
        "Ninja"
    ],
    "cmake.buildDirectory": "${workspaceFolder}/../iree-build",
    "cmakeExplorer.buildDir": "../iree-build",
    "cmakeExplorer.suiteDelimiter": "/",
    "cmakeExplorer.extraCtestRunArgs": "--verbose",
```

Reply:
Fully opened to the idea :) Love it !
I designed the "options" as a starting point for the discussion. 100% opened to add more / remove some.

Now I expect that we won't be able to satisfy everybody's preference. So the way to go is to create a "common ground" where most people are happy with it, and update this directory: `.devcontainer` as little as possible. Which will allow people to make it their own and customize it to their liking.

Reply:
Yep! I'm comfortable at least including Ninja as the preferred generator and setting the test suite delimiter. The build directory setting is useful to standardize for copy/pasting scripts and commands between developers, but some developers (and tools) have a preference for including it as a sibling or child:
```
iree/
  build/  # child, standard practice with CMake projects
  build-android/ # child, we have CI scripts doing this
iree-build/  # sibling, we've had better results with this for development
```

The test suite delimiter gives you a tree view in the extension
![image](https://user-images.githubusercontent.com/4010439/230181245-bb5ea022-6d61-4eec-b6ea-ffa540e93f5c.png)

Reply:
And I like making it a cousin :-)

```
~
  src/
    iree/
  build/
    iree/
      default/
      android/
```